src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -457,27 +457,21 @@ private static void BindProperty(PropertyInfo property, object instance, IConfig
                 // We only support string and enum keys
                 return;
             }
-
+            MethodInfo tryGetValue = dictionaryType.GetMethod("TryGetValue")!;
             PropertyInfo setter = dictionaryType.GetProperty("Item", DeclaredOnlyLookup)!;
             foreach (IConfigurationSection child in config.GetChildren())
             {
+                var key = keyTypeIsEnum ? Enum.Parse(keyType, child.Key) : child.Key;
+                var args = new object?[] { key, null };
+                var _ = tryGetValue.Invoke(dictionary, args);
```

Comment:
```suggestion
                _ = tryGetValue.Invoke(dictionary, args);
```

The difference is your code is declaring a variable named `_`. With the suggestion we are explicitly telling the compiler we are ignoring the result.